### ISSUE TITLE
Build test fails

### ISSUE THREAD

**Issue by oceancolorcoder** [nasa/HyperCP#246]
Application testing on build (merge to master) fails for all 3 platforms, each for its own reason with error

`QWidget: Must construct a QApplication before a QWidget`

appearing for macos and ubuntu at differing fail points and windows showing

`Eforrtl: error (200): program aborting due to control-BREAK event`

**Comment by juanchossn** [comment]
To confrm once cdsapi branch is merged into dev.

**Comment by doizuc** [comment]
It still freezes during the download of ECMWF, I will have a look at it on my fork asap. Extract of logs below.

      File "/Users/runner/work/HyperCP/HyperCP/Source/ProcessL1bTriOS.py", line 389, in processL1b
        modRoot = GetAnc_ecmwf.getAnc_ecmwf(ancGroup)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/Users/runner/work/HyperCP/HyperCP/Source/GetAnc_ecmwf.py", line 199, in getAnc_ecmwf
        ancillary = GetAnc_ecmwf.get_ancillary_main(lat[index], lon[index], lat_timeStamp, ancPath)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/Users/runner/work/HyperCP/HyperCP/Source/GetAnc_ecmwf.py", line 154, in get_ancillary_main
        GetAnc_ecmwf.CAMS_download_ensembles(latEff, lonEff, dateTagEff, timeStampEff, CAMS_variables, pathOut)
      File "/Users/runner/work/HyperCP/HyperCP/Source/GetAnc_ecmwf.py", line 107, in CAMS_download_ensembles
        exit()
      File "<frozen _sitebuiltins>", line 26, in __call__
    SystemExit: None

**Comment by doizuc** [comment]
The **Build & Release Pipeline** passes as noted during our meeting. The issue with assets in the draft release is because the draft release of the previous version wasn't published (looking at HyperCP release page there is no version v1.2.9). If that happens again, you should be able to delete the assets from the previous release while editing the release draft.

Concerning the **Application Testing**, it just failed again on your repo, so I went ahead and tested synchronized [my fork](https://github.com/doizuc/HyperCP/actions/runs/12673338575) and ran the actions and it passed... I suspect there is an issue in the format of the repository secrets. The secrets should contain exactly what the .netrc and .ecmwf_ads_credential files contains. In fact, the action is writing directly the content of the secret to the file (extract of github action below). Unfortunately, you can't see the content of the secret but is the `ECMWF_ADS_CREDENTIALS` secret variable name and content matching the instructions from PR #280 ? Could you set that variable again and try to rerun the test. The variable `NETRC` is fine.

    NETRC: ${{ secrets.NETRC }}
    echo "$NETRC" >> ~/.netrc

    CDSAPIRC: ${{ secrets.ECMWF_ADS_CREDENTIALS }}
    echo "$CDSAPIRC" >> ~/.ecmwf_ads_credentials.json

**Comment by oceancolorcoder** [comment]
This issue is obsolete and will be closed. A new build and application failure issue has been raised.
